production/loki-mixin/alerts.libsonnet

```
@@ -70,6 +70,25 @@
               |||, 'cluster', $._config.per_cluster_label),
             },
           },
+          {
+            // Alert if the compactor has not successfully run compaction in the last 24h.
+            alert: 'LokiCompactorHasNotSuccessfullyRunCompaction',
+            expr: |||
+              # The "last successful run" metric is updated even if the compactor owns no tenants,
+              # so this alert correctly doesn't fire if compactor has nothing to do.
+              (time() - loki_compactor_apply_retention_last_successful_run_timestamp_seconds > 60 * 60 * 24)
```

```suggestion
              (time() - loki_boltdb_shipper_compact_tables_operation_last_successful_run_timestamp_seconds > 60 * 60 * 24)
```

might be better to use the compaction metric instead of the last successful retention run.
metric name is misleading here, it refers to boltdb but it is updated for tsdb indexes as well. naming here needs some clean-up, hopefully by the next release

---

Thanks I might give metrics renaming a go if I have time